tests/unit/test_crt.py

```diff
@@ -164,6 +164,60 @@ def test_delete_request(self):
         self.assertEqual(self.expected_host, crt_request.headers.get("host"))
         self.assertIsNone(crt_request.headers.get("Authorization"))
 
+    def _create_crt_response_error(
+        self, status_code, body, operation_name=None
+    ):
+        return awscrt.s3.S3ResponseError(
+            code=14343,
+            name='AWS_ERROR_S3_INVALID_RESPONSE_STATUS',
+            message='Invalid response status from request',
+            status_code=status_code,
+            headers=[
+                ('x-amz-request-id', 'QSJHJJZR2EDYD4GQ'),
+                (
+                    'x-amz-id-2',
+                    'xDbgdKdvYZTjgpOTzm7yNP2JPrOQl+eaQvUkFdOjdJoWkIC643fgHxdsHpUKvVAfjKf5F6otEYA=',
+                ),
+                ('Content-Type', 'application/xml'),
+                ('Transfer-Encoding', 'chunked'),
+                ('Date', 'Fri, 10 Nov 2023 23:22:47 GMT'),
+                ('Server', 'AmazonS3'),
+            ],
+            body=body,
+            operation_name=operation_name,
+        )
+
+    def test_translate_get_object_404(self):
```

Comment:
It might makes sense to port at least one of these test cases over to the functional test suite to ensure end to end that everything is working fine. Otherwise, we really only have the integration tests which are not run for GitHub CI.

Reply:
Did a quick dive and this is possible but actually requires a bit more scaffolding that I expected. I'm going to vote we punt this part for now since we have an integ test that ensures this works end to end, and come back later to add a proper response-oriented functional test class.